Dockerfile

```diff
@@ -1,22 +1,29 @@
 FROM ubuntu:16.04
 
 RUN apt-get update \
- && apt-get install -y make g++ git \
+ && apt-get install -y make g++ git wget xz-utils \
                        zlib1g-dev libncurses5-dev libssl-dev \
-                       llvm-dev libpcre2-dev \
+                       libpcre2-dev \
  && rm -rf /var/lib/apt/lists/*
 
+RUN wget http://llvm.org/releases/3.9.0/clang+llvm-3.9.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz
+RUN tar xf clang+llvm-3.9.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz
+WORKDIR clang+llvm-3.9.0-x86_64-linux-gnu-ubuntu-16.04
```

Comment:
If you use this path in the following command, you can avoid this `WORKDIR` call.

Also, could you collapse these `RUN` statements together with `&&` so that the `Dockerfile` has fewer layers?  This reduces the overhead when pulling the image.